doc/design/workflow_pacakge.md

```diff
@@ -0,0 +1,85 @@
+# Design Doc: Workflow Package
+
+## Motivation
+
+SQLFlow translates a SQL program, perhaps with extended SQL syntax for AI, into a workflow. Tekton/Argo are Kubernetes native workflow engine when deploying SQLFlow on Kubernetes, SQLFlow leverages Argo/Tekton to do the workflow management.
+
+SQLFlow supports Argo/Tekton as the workflow backend and maybe more in the future. It's different to communicate with the theses workflow engine, .e.g. Argo and Tekton
+are the different CRD on Kubernetes, they are different YAML spec, so it's necessary to orignize a seperate pacakge `workflow` to communicate the workflows with an uniform interface.
```

Comment:
`.e.g. Argo and Tekton are the different ` -> `They are different ..., and they have different YAML spec`